arviz/plots/ecdfplot.py

```diff
@@ -46,26 +57,41 @@ def plot_ecdf(
     values2 : array-like, optional
-        Values to compare to the original sample.
+        deprecated: values to compare to the original sample. Instead use
+        `cdf=scipy.stats.ecdf(values2).cdf.evaluate`.
     cdf : callable, optional
         Cumulative distribution function of the distribution to compare the original sample.
         The function must take as input a numpy array of draws from the distribution.
     difference : bool, default False
         If True then plot ECDF-difference plot otherwise ECDF plot.
-    pit : bool, default False
-        If True plots the ECDF or ECDF-diff of PIT of sample.
-    confidence_bands : bool, default None
-        If True plots the simultaneous or pointwise confidence bands with `1 - fpr`
-        confidence level.
-    pointwise : bool, default False
-        If True plots pointwise confidence bands otherwise simultaneous bands.
+    confidence_bands : str or bool, optional
+        - False: No confidence bands are plotted.
+        - "pointwise": Compute the pointwise (i.e. marginal) confidence band.
+        - True or "simulated": Use Monte Carlo simulation to estimate a simultaneous confidence
+          band.
+        For simultaneous confidence bands to be correctly calibrated, provide `eval_points` that
+        are not dependent on the `values`.
+    band_prob : float, default 0.95
+        The probability that the true ECDF lies within the confidence band. If `confidence_bands`
+        is "pointwise", this is the marginal probability instead of the joint probability.
+    eval_points : array-like, optional
+        The points at which to evaluate the ECDF. If None, `npoints` uniformly spaced points
+        between the data bounds will be used.
     npoints : int, default 100
         This denotes the granularity size of our plot i.e the number of evaluation points
         for the ecdf or ecdf-difference plots.
+    rvs: callable, optional
+        A function that takes an integer `ndraws` and optionally the object passed to
+        `random_state` and returns an array of `ndraws` samples from the same distribution
+        as the original dataset. Required if `method` is "simulated" and variable is discrete.
     num_trials : int, default 500
-        The number of random ECDFs to generate for constructing simultaneous confidence bands.
-    fpr : float, default 0.05
-        The type I error rate s.t `1 - fpr` denotes the confidence level of bands.
+        The number of random ECDFs to generate for constructing simultaneous confidence bands
+        (if `confidence_bands` is "simulated").
+    random_state : {None, int, `numpy.random.Generator`,
+                    `numpy.random.RandomState`}, optional
+        If `None`, the `numpy.random.RandomState` singleton is used. If an `int`, a new
+        ``numpy.random.RandomState`` instance is used, seeded with seed. If a `RandomState` or
+        `Generator` instance, the instance is used.
```

Comment:
I would focus on the argument being passed to `rvs` function, all of what is described only happens for rvs functions that come from scipy distribution classes it isn't really done by any of the ArviZ functions.

Reply:
Can you clarify what change you are suggesting here?

Reply:
ArviZ doesn't really do most of what is described here, behaviour within ArviZ is limited to taking `random_state` and passing it as is to `rvs` function. I would limit the description to that.

Otherwise, if I were to write an rvs function from scratch, I'd make it accept only RandomState of Generator, and expect ArviZ to convert `None` and ints to RandomState before passing the argument to my custom rvs function.

All the exerpt below is scipy behaviour and will be true if `rvs` argument is the method `.rvs` of a scipy distribution but not otherwise, and `rvs` argument isn't limited to `.rvs` method of scipy distributions:

> If `None`, the `numpy.random.RandomState` singleton is used. If an `int`, a new ``numpy.random.RandomState`` instance is used, seeded with seed.

Reply:
Okay, I modified the implementation to support the signature `rvs(ndraws)` if `random_state` is `None` (default). Then it's just documented that one can provide a `random_state` that the provided `rvs` accepts.